lib/elixir/lib/file.ex

```diff
@@ -629,8 +627,6 @@ defmodule File do
         [dest|acc]
       {:error, :eexist} ->
         if callback.(src, dest) do
-          # If rm/1 fails, iF.make_symlink/2 will fail
-          _ = rm(dest)
```

Comment:
Whatever it is not true here, because the symlink will just fail with `{:error, :eexist}` again. Maybe we should remove just the earlier copy. Also can you add some tests please? Thank you!